portfolio/src/main/webapp/slideshow.js

```diff
@@ -12,77 +12,66 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {removeClassName} from "./utils.js";
+import {Utils} from "./utils.js";
 
-/** 
- * Global variables that gives me which 
- * slide I'm currently on in the SlideShow and which one I was in previously.
- */
 class SlideShow {
-  constructor () {
-      /** @private @type {number} */
-      this.slideIndex = 0;
-
-      /** @private @type {number} */
-      this.savedIndex = -1;
+  constructor() {
+    /**
+      * @private @type {number} - This private variable stores the index of 
+      *   the currently rendered slide.
+      */
+    this.currentSlideIndex_ = 0;
   }
 
   /** 
-   * Changes the current slide in a slideshow 
-   * by incrementing by the given n. 
-   * @param {number} slideIncrement - The increment/decrement value for 
+   * If slideIncrement = True, move slideIndex to the right.
+   * Else move slideIndex to the left and call showSlide().
+   * @param {boolean} incrementDirection - The increment/decrement value for 
    *   showing the next slide.
    */
-  showNextSlide(slideIncrement) {
-    this.showSlide(this.slideIndex += slideIncrement);
-  }
-
-  /** 
-   * Display the current nth slide. 
-   * @param {number} current - The current nth index representing the current
-   *   slide.
-   */
-  showCurrentSlide(current) {
-    this.showSlide(this.slideIndex = current);
+  showNextSlide(incrementDirection) {
+    let newIndex = 0;
+    if (incrementDirection) {
```

Comment:
nit: use a ternary operator for simple conditionals like this.

```
const newIndex = incrementDirection ? this.currentSlideIndex_ + 1 : this.currentSlideIndex_ - 1;
```

Reply:
This is awesome! Didn't know you could do that!